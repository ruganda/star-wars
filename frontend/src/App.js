import logo from './logo.svg';
import './App.css';
import PeopleList from './components/people';
import PeopleContextProvider from './contexts/peopleContext';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link
} from "react-router-dom";
import ProfileDetails from './components/profileDetails';


function App() {
  return (

    <Router>
    <Switch>
      <PeopleContextProvider>
        <Route path="/people" exact component={PeopleList} />
        <Route path="/people/:name" exact={true} component={ProfileDetails} />
      </PeopleContextProvider>
    </Switch>
    </Router>


    // <div className="App">
    //   <PeopleContextProvider>
    //     <PeopleList/>
    //   </PeopleContextProvider>
    // </div>
  );
}

export default App;
