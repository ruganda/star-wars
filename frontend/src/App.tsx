import './App.css';
import PeopleList from './people';
import PeopleContextProvider from './contexts/peopleContext';
import {
  BrowserRouter as Router,
  Switch,
  Route,

} from "react-router-dom";
import ProfileDetails from './components/profileDetails';


function App() {
  return (

    <Router>
    <Switch>
      <PeopleContextProvider>
        <Route path="/people" exact component={PeopleList} />
        <Route path="/people/:name" exact={true} component={ProfileDetails} />
      </PeopleContextProvider>
    </Switch>
    </Router>
  );
}

export default App;
