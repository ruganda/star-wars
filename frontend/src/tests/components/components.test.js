import App from "../../App";
import TestRenderer from "react-test-renderer";
import { MockedProvider } from "@apollo/client/testing";
import { peopleMocks } from "../mock";
import PeopleContextProvider from "../../contexts/peopleContext";

import PeopleTable from "../../components/peopleTable";
import PaginationComponent from "../../components/pagination";
import ProfileDetails from "../../components/profileDetails";
import SearchForm from "../../components/searchForm";

const {act} = TestRenderer;


describe("PeopleTable", () => {
  it("renders the table component without crashing", () => {
    const component = TestRenderer.create(
      <MockedProvider mocks={peopleMocks} addTypename={false}>
        <PeopleContextProvider>
          <PeopleTable data={[]} />
        </PeopleContextProvider>
      </MockedProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("<Pagination />", () => {
  it("renders <PaginationComponent/> without crashing>", () => {
    const component = TestRenderer.create(
      <MockedProvider mocks={peopleMocks} addTypename={false}>
        <App />
      </MockedProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("<PaginationComponent/>", () => {
  it("renders <PaginationComponent/> without crashing", () => {
    const component = TestRenderer.create(
      <MockedProvider mocks={peopleMocks} addTypename={false}>
        <PaginationComponent />
      </MockedProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});

describe("<ProfileDetails/>", () => {
  it("renders <ProfileDetails /> without crashing", () => {
    const props = {
      match: {
        params: {
          name: "Luke Skywalker",
        },
      },
    };
    const component = TestRenderer.create(
      <MockedProvider mocks={peopleMocks} addTypename={false}>
        <PeopleContextProvider>
          <ProfileDetails {...props}/>
        </PeopleContextProvider>
      </MockedProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("<SearchForm/>", () => {
  it("renders <SearchForm/> without crashing", () => {
    const component = TestRenderer.create(
      <MockedProvider mocks={peopleMocks} addTypename={false}>
        <SearchForm />
      </MockedProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should search by name in the  <SearchForm/>", () => {
    const component = TestRenderer.create(
      <MockedProvider mocks={peopleMocks} addTypename={false}>
        <SearchForm />
      </MockedProvider>
    );
    const event ={
      preventDefault: jest.fn(),
      target: {
        value: "Luke Skywalker",
      }
    }
    act(() => {
      component.root.findByType('input').props.onChange(event);
    });
    

    const input = component.root.findByType('input')
    expect(input.props.value).toEqual("Luke Skywalker");    
  });

  it("should submit a search in the  <SearchForm/>", () => {

    const component = TestRenderer.create(
      <MockedProvider mocks={peopleMocks} addTypename={false}>
        <PeopleContextProvider>
          <SearchForm />
        </PeopleContextProvider>
      </MockedProvider>

    );

    const event ={
      preventDefault: jest.fn()
    }
    act(() => {
      component.root.findByType('form').props.onSubmit(event);
    });
    
    const input = component.root.findByType('input')
    // the input is ""because we clear the input after submitting
    expect(input.props.value).toEqual("");    
  });
});
