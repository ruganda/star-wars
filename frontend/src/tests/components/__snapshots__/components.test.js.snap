// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> renders <> 1`] = `null`;

exports[`<Pagination /> renders <PaginationComponent without > 1`] = `null`;

exports[`<Pagination /> renders <PaginationComponent/> without crashing> 1`] = `null`;

exports[`<Pagination /> renders <PaginationCopmonent > 1`] = `null`;

exports[`<Pagination /> renders learn react link  1`] = `null`;

exports[`<Pagination /> renders learn react link 1`] = `null`;

exports[`<PaginationComponent/> renders <PaginationComponent/> without 1`] = `
<div
  className="makeStyles-rootContainer-4"
>
  <nav
    aria-label="pagination navigation"
    className="MuiPagination-root"
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
        "width": "100VW",
      }
    }
  >
    <ul
      className="MuiPagination-ul"
    >
      <li>
        <button
          aria-label="Go to previous page"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiPaginationItem-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          aria-label="Go to next page"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiPaginationItem-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </button>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`<PaginationComponent/> renders <PaginationComponent/> without crashing 1`] = `
<div
  className="makeStyles-rootContainer-4"
>
  <nav
    aria-label="pagination navigation"
    className="MuiPagination-root"
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
        "width": "100VW",
      }
    }
  >
    <ul
      className="MuiPagination-ul"
    >
      <li>
        <button
          aria-label="Go to previous page"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiPaginationItem-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          aria-label="Go to next page"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiPaginationItem-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </button>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`<ProfileDetails/> renders <ProfileDetails /> without 1`] = `
<div
  className="makeStyles-profileContainer-8"
>
  <h2>
     Profile Page
  </h2>
  <div
    className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <span
      className="MuiSkeleton-root MuiSkeleton-circle MuiSkeleton-pulse"
      style={
        Object {
          "height": 200,
          "width": 200,
        }
      }
    />
    <div
      style={
        Object {
          "width": "300px",
        }
      }
    >
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`<ProfileDetails/> renders <ProfileDetails /> without crashing 1`] = `
<div
  className="makeStyles-profileContainer-8"
>
  <h2>
     Profile Page
  </h2>
  <div
    className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <span
      className="MuiSkeleton-root MuiSkeleton-circle MuiSkeleton-pulse"
      style={
        Object {
          "height": 200,
          "width": 200,
        }
      }
    />
    <div
      style={
        Object {
          "width": "300px",
        }
      }
    >
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`<SearchForm/> renders <SearchForm/> without 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink={false}
      htmlFor="outlined-search"
      id="outlined-search-label"
    >
      Search name
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputTypeSearch"
        disabled={false}
        id="outlined-search"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="search"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="PrivateNotchedOutline-legendLabelled-12"
        >
          <span>
            Search name
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-searchButton-9"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="submit"
  >
    <span
      className="MuiButton-label"
    >
      Search
    </span>
  </button>
</form>
`;

exports[`<SearchForm/> renders <SearchForm/> without crashing 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink={false}
      htmlFor="outlined-search"
      id="outlined-search-label"
    >
      Search name
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputTypeSearch"
        disabled={false}
        id="outlined-search"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="search"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="PrivateNotchedOutline-legendLabelled-12"
        >
          <span>
            Search name
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-searchButton-9"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="submit"
  >
    <span
      className="MuiButton-label"
    >
      Search
    </span>
  </button>
</form>
`;

exports[`<SearchForm/> should  <SearchForm/> without crashing 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink={false}
      htmlFor="outlined-search"
      id="outlined-search-label"
    >
      Search name
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputTypeSearch"
        disabled={false}
        id="outlined-search"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="search"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="PrivateNotchedOutline-legendLabelled-12"
        >
          <span>
            Search name
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-searchButton-9"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="submit"
  >
    <span
      className="MuiButton-label"
    >
      Search
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</form>
`;

exports[`<SearchForm/> should search by name in the  <SearchForm/> 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink={false}
      htmlFor="outlined-search"
      id="outlined-search-label"
    >
      Search name
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputTypeSearch"
        disabled={false}
        id="outlined-search"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="search"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="PrivateNotchedOutline-legendLabelled-12"
        >
          <span>
            Search name
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-searchButton-9"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="submit"
  >
    <span
      className="MuiButton-label"
    >
      Search
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</form>
`;

exports[`<SearchForm/> should search by name in the  <SearchForm/> without crashing 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink={false}
      htmlFor="outlined-search"
      id="outlined-search-label"
    >
      Search name
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputTypeSearch"
        disabled={false}
        id="outlined-search"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="search"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="PrivateNotchedOutline-legendLabelled-12"
        >
          <span>
            Search name
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-searchButton-9"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="submit"
  >
    <span
      className="MuiButton-label"
    >
      Search
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</form>
`;

exports[`PeopleTable renders the table component without crashing 1`] = `
<div
  className="MuiPaper-root MuiTableContainer-root makeStyles-rootContainer-2 MuiPaper-elevation1 MuiPaper-rounded"
>
  <table
    aria-label="a dense table"
    className="MuiTable-root makeStyles-table-1"
    role={null}
  >
    <thead
      className="MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        >
          Name
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        >
          Gender
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        >
          Height
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        >
          Mass
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        >
          Homeworld
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
      role={null}
    />
  </table>
</div>
`;
